agasc/agasc.py

```diff
@@ -135,20 +134,15 @@ def add_pmcorr_columns(stars, date):
     # field to degrees.  The AGASC PM is specified in milliarcsecs / year, so this
     # is dyear * (degrees / milliarcsec)
 
-    # The dyear for proper motion is only relevant for stars that have defined proper motion
-    # so set to zero for all stars by default
-    dyear = np.zeros(len(stars))
-    has_pm = (stars['PM_DEC'] != -9999) | (stars['PM_RA'] != -9999)
-    # For most of them, the epoch is fixed at 2000, so we don't need N calls to DateTime to
-    # figure that out
-    epoch_is_2000 = (stars['EPOCH'] == 2000.0)
-    ok = has_pm & epoch_is_2000
-    dyear[ok] = (DateTime(date) - DateTime(2000, format='frac_year')) / 365.25
-    # For stars with proper motion correction but epoch != 2000, calculate individually.
-    ok = has_pm & ~epoch_is_2000
-    dyear[ok] = (DateTime(date) - DateTime(stars['EPOCH'][ok], format='frac_year')) / 365.25
-    pm_to_degrees = dyear / (3600. * 1000.)
+    if np.asarray(date).shape == ():
```

Comment:
@taldcroft Would you please remind me of the expected types and shapes of date?  It isn't clear to me which problem this is code change is solving.

Reply:
Ah, I see.  The new get_stars() routine repurposes add_pmcorr_columns() to receive date possibly as a thing with the same length as stars.  I think add_pmcorr_columns probably needs a real docstring now and maybe a rename of date to dates in the args.

Reply:
Documentation update done.  I left the arg as `date` on the off chance this non-public method is being used directly somewhere.